charts/everest/charts/common/templates/_upgrade_checks.yaml.tpl

```diff
@@ -21,18 +63,31 @@ spec:
           name: {{ $hookName }}
           command:
             - /bin/sh
-            - -c
+            - -ec
             - |
               OS=$(uname -s | tr '[:upper:]' '[:lower:]')
               ARCH=$(uname -m)
+
+              # Map ARCH to match available builds
+              if [ "$ARCH" = "x86_64" ]; then
+                  ARCH="amd64"
+              elif [ "$ARCH" = "aarch64" ]; then
+                  ARCH="arm64"
+              else
+                  echo "Unsupported architecture: $ARCH"
+                  exit 1
+              fi
+
               VERSION={{ .version }}
               apk add --no-cache --quiet curl
               curl -sSL -o everestctl https://github.com/percona/everest/releases/download/v${VERSION}/everestctl-${OS}-${ARCH}
               chmod -R 777 ./everestctl
               
               echo "Checking requirements for upgrade to version ${VERSION}"
-              ./everestctl upgrade --dry-run --version-metadata-url={{ .versionMetadataURL }}
+              ./everestctl upgrade --in-cluster --dry-run --version-metadata-url={{ .versionMetadataURL }} --kubeconfig=""
```

Comment:
There is an issue in the CLI where it tries to read kubeconfig from `~/.kube/config` even though `--in-cluster` is specified. Setting `--kubeconfig=""` resolves it. 

I know its a bit weird, but I'm doing it this way for now to unblock testing.. I will fix it in the CLI later